tests/test_hooks_intelligence.py

```
@@ -0,0 +1,422 @@
+
+# ── context_inject ──
+
+from gradata.hooks.context_inject import main as context_main
+
+
+def test_context_inject_returns_context(tmp_path):
+    mock_brain = MagicMock()
+    mock_brain.search.return_value = [{"text": "Relevant brain knowledge here"}]
+
+    with patch.dict(os.environ, {"GRADATA_BRAIN_DIR": str(tmp_path)}):
+        with patch("gradata.brain.Brain", return_value=mock_brain):
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)